cdap-app-fabric/src/main/java/io/cdap/cdap/internal/messaging/DelegatingMessagingService.java

```diff
@@ -130,6 +135,11 @@ private MessagingService getDelegate() {
             "Unsupported messaging service implementation " + getName());
       }
       LOG.info("Messaging service {} is loaded", messagingService.getName());
+      try {
+        messagingService.initialize(new DefaultMessagingContext(this.cConf));
```

Comment:
Why move here instead of doing it in the regular initialize method?

Reply:
Messaging currently doesn't have initialize(). 
We need this to create spanner database clients. Extension is loaded at this particular place and the current object's delegate is set to the messagingService. So thought this would be an appropriate place for initialize().

Reply:
+1